latch_cli/services/sync.py

```diff
@@ -345,4 +353,7 @@ def sync(
             )
         click.echo()
 
-    sync_rec(srcs, dest, delete=delete)
+    with ProcessPoolExecutor(max_workers=get_max_workers()) as executor:
+        futures = sync_rec(srcs, normalize_path(dest), delete=delete, executor=executor)
+        for future in as_completed(futures):
+            future.result()
```

Comment:
```suggestion
        sync_rec(srcs, normalize_path(dest), delete=delete, executor=executor)
```

`executor.__exit__()` will act like calling `executor.shutdown(wait=True)` which will wait for all the outstanding futures in the executor (see https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.shutdown)

Reply:
this way you also don't have to change the return type of the fn

Reply:
Goo comment.